backend/jvm/src/main/scala/zio/logging/backend/PlatformSpecificBackends.scala

```diff
@@ -0,0 +1,69 @@
+package zio.logging.backend
+
+import org.slf4j.{ LoggerFactory, MDC }
+import zio.logging.{ LogFormat, logAnnotation }
+import zio.{ FiberId, LogLevel, LogSpan, RuntimeConfigAspect, ZFiberRef, ZLogger, ZTraceElement }
+
+import scala.jdk.CollectionConverters._
+
+trait PlatformSpecificBackends {
+
+  def slf4j(
+    logLevel: zio.LogLevel,
+    format: LogFormat[String],
+    rootLoggerName: ZTraceElement => String
+  ): RuntimeConfigAspect =
+    RuntimeConfigAspect.addLogger(slf4jLogger(rootLoggerName, logLevel, format))
+
+  def slf4j(
+    logLevel: zio.LogLevel,
+    format: LogFormat[String]
+  ): RuntimeConfigAspect =
+    slf4j(logLevel, format, _ => "zio-slf4j-logger")
+
+  def slf4j(
+    logLevel: zio.LogLevel
+  ): RuntimeConfigAspect =
+    slf4j(logLevel, LogFormat.default, _ => "zio-slf4j-logger")
+
+  private def slf4jLogger(
+    rootLoggerName: ZTraceElement => String,
+    logLevel: LogLevel,
+    format: LogFormat[String]
+  ): ZLogger[Unit] =
+    new ZLogger[Unit] {
+      val formatLogger: ZLogger[Option[String]] = format.toLogger.filterLogLevel(_ >= logLevel)
+
+      override def apply(
+        trace: ZTraceElement,
+        fiberId: FiberId,
+        logLevel: LogLevel,
+        message: () => String,
+        context: Map[ZFiberRef.Runtime[_], AnyRef],
+        spans: List[LogSpan]
+      ): Unit =
+        formatLogger(trace, fiberId, logLevel, message, context, spans).foreach { message =>
+          val slf4jLogger = LoggerFactory.getLogger(rootLoggerName(trace))
+
+          val previous =
+            context.get(logAnnotation) match {
+              case Some(annotations: Map[String, String] @unchecked) if annotations.nonEmpty =>
+                MDC.setContextMap(annotations.asJava)
+                Some(Option(MDC.getCopyOfContextMap).getOrElse(java.util.Collections.emptyMap[String, String]()))
```

Comment:
This is copying the context map AFTER it has updated it to the log annotations. If you are trying to restore it to the old value, then I would do that prior to setting the context map.

Reply:
my bad! I still need to add test for this